server/src/main/java/org/elasticsearch/search/aggregations/metrics/CardinalityAggregator.java

```diff
@@ -288,29 +289,30 @@ public void collect(int doc, long bucketOrd) throws IOException {
 
         @Override
         public void postCollect() throws IOException {
-            final FixedBitSet allVisitedOrds = new FixedBitSet(maxOrd);
-            for (long bucket = visitedOrds.size() - 1; bucket >= 0; --bucket) {
-                final FixedBitSet bits = visitedOrds.get(bucket);
-                if (bits != null) {
-                    allVisitedOrds.or(bits);
+            try (BitArray allVisitedOrds = new BitArray(maxOrd, bigArrays)) {
+                for (long bucket = visitedOrds.size() - 1; bucket >= 0; --bucket) {
+                    final BitArray bits = visitedOrds.get(bucket);
+                    if (bits != null) {
+                        allVisitedOrds.or(bits);
+                    }
                 }
-            }
 
-            final MurmurHash3.Hash128 hash = new MurmurHash3.Hash128();
-            try (LongArray hashes = bigArrays.newLongArray(maxOrd, false)) {
-                for (int ord = allVisitedOrds.nextSetBit(0); ord < DocIdSetIterator.NO_MORE_DOCS;
-                        ord = ord + 1 < maxOrd ? allVisitedOrds.nextSetBit(ord + 1) : DocIdSetIterator.NO_MORE_DOCS) {
-                    final BytesRef value = values.lookupOrd(ord);
-                    MurmurHash3.hash128(value.bytes, value.offset, value.length, 0, hash);
-                    hashes.set(ord, hash.h1);
-                }
+                try (LongArray hashes = bigArrays.newLongArray(maxOrd, false)) {
+                    final MurmurHash3.Hash128 hash = new MurmurHash3.Hash128();
+                    for (long ord = allVisitedOrds.nextSetBit(0); ord < DocIdSetIterator.NO_MORE_DOCS;
```

Comment:
Should it use `Long.MAX_VALUE` instead of `DocIdSetIterator.NO_MORE_DOCS` ?

Reply:
Yes